packages/web-app-epub-reader/tests/unit/app.spec.ts

```diff
@@ -1,23 +1,197 @@
+  chaptersSelect: '.epub-reader-controls-chapters-select',
+  navigateLeft: '.epub-reader-navigate-left',
+  navigateRight: '.epub-reader-navigate-right'
+}
+describe('Epub reader app', () => {
+  it('renders correctly', () => {
+    const { wrapper } = getWrapper()
     expect(wrapper.html()).toMatchSnapshot()
   })
+  describe('theme', () => {
+    it('sets the theme based on current theme setting', async () => {
+      const { wrapper } = getWrapper({ localStorage: { fontSizePercentage: 50 } })
+      await nextTicks(2)
+      expect(wrapper.vm.rendition.themes.select).toHaveBeenCalledWith('light')
+      expect(wrapper.vm.rendition.themes.fontSize).toHaveBeenCalledWith('50%')
+    })
+  })
+  describe('font size', () => {
+    it('initializes with default font size percentage', () => {
+      const { wrapper } = getWrapper()
+      expect(wrapper.vm.currentFontSizePercentage).toBe(100)
+    })
+    it('initializes with local storage font size when set', () => {
+      const { wrapper } = getWrapper({ localStorage: { fontSizePercentage: 50 } })
+      expect(wrapper.vm.currentFontSizePercentage).toBe(50)
+    })
+    describe('increase font size button', () => {
+      it('increases font size when clicked', async () => {
+        const { wrapper } = getWrapper()
+        await nextTicks(2)
+        await wrapper.find(selectors.increaseFontSize).trigger('click')
+        expect(wrapper.vm.currentFontSizePercentage).toBe(110)
+      })
+      it('is disabled when "MAX_FONT_SIZE_PERCENTAGE" is reached', () => {
+        const { wrapper } = getWrapper({ localStorage: { fontSizePercentage: 150 } })
+        expect((wrapper.find(selectors.increaseFontSize) as any).isDisabled()).toBe(true)
+      })
+    })
+    describe('decrease font size button', () => {
+      it('decreases font size when clicked', async () => {
+        const { wrapper } = getWrapper()
+        await nextTicks(2)
+        await wrapper.find(selectors.decreaseFontSize).trigger('click')
+        expect(wrapper.vm.currentFontSizePercentage).toBe(90)
+      })
+      it('is disabled when "MIN_FONT_SIZE_PERCENTAGE" is reached', () => {
+        const { wrapper } = getWrapper({ localStorage: { fontSizePercentage: 50 } })
+        expect((wrapper.find(selectors.decreaseFontSize) as any).isDisabled()).toBe(true)
+      })
+    })
+    describe('reset font size button', () => {
+      it('resets font size when clicked', async () => {
+        const { wrapper } = getWrapper({ localStorage: { fontSizePercentage: 50 } })
+        await nextTicks(2)
+        await wrapper.find(selectors.resetFontSize).trigger('click')
+        expect(wrapper.vm.currentFontSizePercentage).toBe(100)
+      })
+      it('shows the current font size', async () => {
+        const { wrapper } = getWrapper()
+        await nextTicks(2)
+        await wrapper.find(selectors.decreaseFontSize).trigger('click')
+        expect(wrapper.find(selectors.resetFontSize).text()).toBe('90%')
+      })
+    })
+  })
```

Comment:
Shouldn't these tests use `expect(wrapper.vm.rendition.themes.fontSize).toHaveBeenCalledWith(…)` as well instead of expect(wrapper.vm.currentFontSizePercentage).toBe(…)?

Reply:
I think both ? What  do you think ?

Reply:
Hm - I think the value of the ref is an implementation detail.

Checking `wrapper.vm.rendition.themes.fontSize` and a button value (or wherever it is displayed) seems like a better approach to me than checking the ref's value